automated install/basic-install.sh

```diff
@@ -88,16 +88,27 @@ if command -v apt-get &> /dev/null; then
   DNSMASQ_USER="dnsmasq"
 
   test_dpkg_lock() {
-    i=0
-    while fuser /var/lib/dpkg/lock >/dev/null 2>&1 ; do
-      sleep 0.5
-      ((i=i+1))
+    # Account for cloud init type installs that autorun apt on boot
+    local maxtries
+    local retVal
+    maxtries=75
+
+    for i in $(seq 1 $maxtries);
+    do
+      "${UPDATE_PKG_CACHE}"
+      retVal=$?
+      # if we find lock condition
+      if [[ $retVal -eq 100 ]]; then
```

Comment:
Why no `else` condition? I could image a number of errors that could happen (like one or more mirrors not available, etc.)